### PR TITLE
Add attachEmotes method + cleanup

### PR DESCRIPTION
**Pull request by JamyDev**
You can ignore most of the json diff, that was just linting

### CODE REVIEW

**Comment by JamyDev** [comment]
https://github.com/WatchBeam/developers/pull/120/files#diff-219a39d1eb178df66090482ccfb65ab6R305

File: src/reference/chat/data.json [hunk 1]
```diff
@@ -533,7 +649,10 @@
           "user": {
             "user_name": "username",
             "user_id": 102534,
-            "user_roles": ["Pro", "User"]
+            "user_roles": [
+              "Pro",
+              "User"
+            ]
```
Comment (connor4312) [inline src/reference/chat/data.json]: Can we just pass this file though `jq` or some standard formatter as part of the build process so we don't keep changing it? :D

File: src/reference/chat/data.json [hunk 2]
```diff
@@ -271,6 +301,69 @@
           "id": 12
         }
       }
+    },
+    "attachEmotes": {
+      "description": "Enables an enhancement to the `ChatMessage` event. Will populate the `meta` property of the event with a hash containing the emoticon text mapped to the base64 PNG representation.",
+      "examples": [
+        {
+          "description": "To enable, call the api method.",
```
Comment (connor4312) [inline src/reference/chat/data.json]: `Tells the chat server that you want emotes to be attached to chat messages.`

File: src/reference/chat/data.json (hunk 1 above)
Comment (SimonSchick) [inline src/reference/chat/data.json]: I'd just pass it through `JSON.stringify(JSON.parse(...), null, '    ')`

**Comment by JamyDev** [comment]
Did the linting, I know it can be a tad more gulpier. So let me know

File: build/misc.js [hunk 3]
```diff
@@ -211,4 +211,15 @@ module.exports = (gulp, $) => {
         .pipe($.if(config.minify, $.uglify()))
         .pipe(gulp.dest(config.dist.js));
     });
+
+    gulp.task('lint-json', () => {
+        const files = [
```
Comment (SimonSchick) [inline build/misc.js]: could use gulp src instead, but this works just as well.

**Pull request merged**